File: nuxt3_app/model/EntidadInfo.ts
export class EntidadInfo {
    protected fechaCreacion?: Date;
    protected fechaModificacion?: Date;
    protected usuarioCreacion?: string;
    protected usuarioModificacion?: string;
    protected valido?: boolean = true;
  
    constructor(
      fechaCreacion?: Date,
      fechaModificacion?: Date,
      usuarioCreacion?: string,
      usuarioModificacion?: string,
      valido?: boolean
    ) {
      this.fechaCreacion = fechaCreacion;
      this.fechaModificacion = fechaModificacion;
      this.usuarioCreacion = usuarioCreacion;
      this.usuarioModificacion = usuarioModificacion;
      this.valido = valido;
    }
  
    toJson(): string {
      return JSON.stringify({
        fechaCreacion: this.fechaCreacion,
        fechaModificacion: this.fechaModificacion,
        usuarioCreacion: this.usuarioCreacion,
        usuarioModificacion: this.usuarioModificacion,
        valido: this.valido,
      });
    }
  }
  

File: microservices/backend-nuxt-krakenD/src/main/java/ea/ciges/krakend/model/EntidadInfo.java
package ea.ciges.krakend.model;


import jakarta.persistence.MappedSuperclass;
import lombok.Getter;
import lombok.Setter;

import java.io.Serializable;
import java.time.LocalDateTime;

@MappedSuperclass
@Setter
@Getter
public abstract class EntidadInfo<E extends Serializable> extends EntidadPK<E> {



    protected LocalDateTime fechaCreacion;

    protected LocalDateTime fechaModificacion;

    protected String usuarioCreacion;

    protected String usuarioModificacion;

    protected Boolean valido = true;


}
